node/src/cli.rs

```diff
@@ -16,9 +16,70 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-use sc_cli::{KeySubcommand, RunCmd, SignCmd, VanityCmd, VerifyCmd};
+use sc_cli::{KeySubcommand, SignCmd, VanityCmd, VerifyCmd};
+use std::{fmt, str::FromStr};
 use structopt::StructOpt;
 
+#[derive(Debug, Clone)]
+pub struct RuntimeInstanceError(String);
+
+impl fmt::Display for RuntimeInstanceError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        let RuntimeInstanceError(message) = self;
+        write!(f, "RuntimeInstanceError: {}", message)
+    }
+}
+
+#[derive(Debug, StructOpt)]
+pub enum RuntimeInstance {
+    Main,
+    Staking,
+}
+
+impl RuntimeInstance {
+    fn variants() -> [&'static str; 2] {
+        ["main", "staking"]
+    }
+
+    pub fn is_staking_runtime(&self) -> bool {
+        match self {
+            Self::Main => false,
+            Self::Staking => true,
+        }
+    }
+}
+
+impl fmt::Display for RuntimeInstance {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match *self {
+            Self::Main => write!(f, "main"),
+            Self::Staking => write!(f, "staking"),
+        }
+    }
+}
+
+impl Default for RuntimeInstance {
+    fn default() -> Self {
+        RuntimeInstance::Main
+    }
+}
+
+impl FromStr for RuntimeInstance {
```

Comment:
strum would also provide all the conversions to and from String for your enum. So you can remove this code if you use that.

Reply:
☝🏻 I'd forgotten about [`strum`](https://docs.rs/strum/0.22.0/strum/). I second the recommendation of choosing a library to simplify manual implementations.

Reply:
as long as it supports `no_std` that's a great idea!

Reply:
As suggested, extra cli options removed